### DISCUSSION TITLE
Block-based mode

### DISCUSSION THREAD

**Discussion by johnlunney** [zk-org/zk#241] [Ideas]
I love the minimal approach here. Did you ever consider a block-based mode? 
This would be like Logseq, where the basic unit is a bullet or paragraph, rather than the file.
So if there is a back-link or tag, it points to the exact bullet it was used in. https://youtu.be/cy5A-_S1bnU is a good explanation.

The benefit (from my perspective) is that in just writing your daily journal and mentioning other nodes, each bullet is building up those pages, and blocks can be embedded in other notes.

Having looked through the source and data structures here, there'd be a fair amount to change, not impossible. 
I'm thinking of writing a simple proof-of-concept, but thought I'd mention here as well!

**Comment by mickael-menu** [comment]
Block-based notes are an interesting concept, especially when outlining or summarising. I have some concerns though about implementing this in `zk`:

1. I'm not sure how this could be implemented without a custom dedicated UI app? How did you imagine using this with the `zk` CLI interface?
2. A core tenet of `zk` is to use a standard interoperable format to make sure the notebook is future-proof and not locking in users to this app. However, there's some precedence for embedding notes, [Obsidian is using `![[Embedded note]`](https://help.obsidian.md/How+to/Embed+files) which should at least work as regular links in an app supporting Wiki links.
